c4builder/src/3 Component Diagrams/3 Rental service/rental-service.puml

```diff
@@ -0,0 +1,43 @@
+@startuml
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+!define DEVICONS https://raw.githubusercontent.com/tupadr3/plantuml-icon-font-sprites/master/devicons
+!include <logos/rabbitmq.puml>
+
+!include DEVICONS/dotnet.puml
+!include DEVICONS/postgresql.puml
+
+LAYOUT_WITH_LEGEND()
+
+title Component diagram for eLibraryOnContainers - Rental service
+
+Container(gateway, "API Gateway", ".NET with ASP.NET Core", "Routes all requests to appropriate services endpoints", $sprite="dotnet")
+ContainerQueue(rabbit, "Event bus", "RabbitMQ", "Event bus for asynchronous communication", $sprite=rabbitmq)
+ContainerDb(rentals_db, "Rentals database", "PostgreSQL", "Holds rentals events and physical copies information", $sprite="postgresql")
+ContainerDb(rentals_read_db, "Rentals (READ) database", "PostgreSQL", "Holds aggregated rentals and physical copies availability information", $sprite="postgresql")
+
+Container_Boundary(rentals, "Rental service") {
+  Component(rentalController, "Rentals Controller", "Rest Controller", "Allows users to send requests related to book renting.")
+  Component(rentalService, "Rentals Service", "Events publisher", "Produces rentals events.")
+  Component(rentalReader, "Rentals Reader", "Spring service", "Reads and aggregates events from database.")
```

Comment:
`Rentals Reader` doesn't aggregate and/or read events. `Rentals Worker` is responsible for aggregation (projection) and saving projection to READ database. `Rentals Reader` reads projections from READ database.